core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -120,7 +124,7 @@ public static boolean isTimestamptz(Schema schema) {
   }
 
   public static boolean isOptionSchema(Schema schema) {
-    if (schema.getType() == UNION && schema.getTypes().size() == 2) {
+    if (schema.getType() == UNION && schema.getTypes().size() >= 2) {
```

Comment:
This doesn't handle the case where a union is `["int", "float", "null"]`. What about keeping union-as-struct handling in separate methods? That way existing uses don't need to account for different behavior depending on whether a schema is a union or an option. It's concerning that this changes the contract of this method for existing callers.

Reply:
let add handling to return false when its UNION and first element is not null when size > 2

Reply:
fixed keeping existing behaviour for non union types